File: apps/flutter/lib/models/service.dart
class ServiceModel {
  final String id;
  final String name;
  final String? description;
  final double price;
  final double? originalPrice;
  final String? coverImage;
  final List<String>? images;
  final String? categoryId;
  final String? categoryName;
  final int? duration;
  final int? salesCount;
  final List<ServiceSpec>? specs;
  final String? createdAt;

  ServiceModel({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    this.originalPrice,
    this.coverImage,
    this.images,
    this.categoryId,
    this.categoryName,
    this.duration,
    this.salesCount,
    this.specs,
    this.createdAt,
  });

  factory ServiceModel.fromJson(Map<String, dynamic> json) {
    final category = json['category'];
    return ServiceModel(
      id: json['id']?.toString() ?? '',
      name: json['name'] ?? '',
      description: json['description'],
      price: (json['price'] is int)
          ? (json['price'] as int).toDouble()
          : json['price']?.toDouble() ?? 0.0,
      originalPrice: (json['original_price'] is int)
          ? (json['original_price'] as int).toDouble()
          : json['original_price']?.toDouble(),
      coverImage: json['cover_image'],
      images: json['images'] != null ? List<String>.from(json['images']) : null,
      categoryId: json['category_id']?.toString(),
      categoryName: category is Map ? category['name']?.toString() : null,
      duration: json['duration'],
      salesCount: json['sales_volume'],
      specs: json['specs'] != null
          ? (json['specs'] as List).map((e) => ServiceSpec.fromJson(e)).toList()
          : null,
      createdAt: json['created_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'original_price': originalPrice,
      'cover_image': coverImage,
      'images': images,
      'category_id': categoryId,
      'category_name': categoryName,
      'duration': duration,
      'sales_volume': salesCount,
      'specs': specs?.map((e) => e.toJson()).toList(),
      'created_at': createdAt,
    };
  }
}

class ServiceSpec {
  final String name;
  final String? value;
  final double? priceAdjust;

  ServiceSpec({required this.name, this.value, this.priceAdjust});

  factory ServiceSpec.fromJson(Map<String, dynamic> json) {
    return ServiceSpec(
      name: json['name'] ?? '',
      value: json['value'],
      priceAdjust: (json['price_adjust'] is int)
          ? (json['price_adjust'] as int).toDouble()
          : json['price_adjust']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'value': value,
      'price_adjust': priceAdjust,
    };
  }
}

File: service/app/model/Service.php
<?php
// Copyright (c) 2026 erik <erik@erik.xyz> — https://erik.xyz

declare(strict_types=1);

namespace app\model;

use Erikwang2013\WebmanScout\Searchable;
use Illuminate\Database\Eloquent\SoftDeletes;
use support\Model;

class Service extends Model
{
    use Searchable, SoftDeletes;

    protected $table = 'appointment_service';
    public $incrementing = false;
    protected $keyType = 'string';
    public $timestamps = true;

    protected $fillable = [
        'category_id', 'name', 'description', 'cover_image', 'images',
        'price', 'original_price', 'duration', 'specs',
        'sales_volume', 'sort', 'status',
    ];

    protected $casts = [
        'images' => 'array',
        'specs' => 'array',
        'price' => 'float',
        'original_price' => 'float',
        'duration' => 'integer',
        'sales_volume' => 'integer',
        'sort' => 'integer',
        'status' => 'integer',
    ];

    protected $hidden = ['deleted_at'];

    /**
     * 搜索引擎索引名称
     */
    public function searchableAs(): string
    {
        return 'appointment_services';
    }

    /**
     * 获取索引数据数组
     */
    public function toSearchableArray(): array
    {
        return [
            'id' => $this->id,
            'category_id' => $this->category_id,
            'name' => $this->name,
            'description' => $this->description,
            'price' => $this->price,
            'original_price' => $this->original_price,
            'duration' => $this->duration,
            'sales_volume' => $this->sales_volume,
            'sort' => $this->sort,
            'status' => $this->status,
        ];
    }

    public function category()
    {
        return $this->belongsTo(ServiceCategory::class, 'category_id');
    }

    public function reviews()
    {
        return $this->hasMany(OrderReview::class, 'service_id');
    }

}
